### DISCUSSION TITLE
Help needed - 'tuple' object has no attribute 'pk'

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#7456] [Q&A] [answered]
Hey,

I have a Prefix container 192.168.202.0/24
with several smaller child prefixes 192.168.202.0/26, 192.168.202.64/26 and so on.

When I open the details of the 192.168.202.0/24 and go to the IPAddresses tab, I get a traceback
```
Django Version:	3.2.7
Exception Type:	AttributeError
Exception Value:	
'tuple' object has no attribute 'pk'
Exception Location:	/opt/netbox/netbox/utilities/tables.py, line 131, in <genexpr>
Python Executable:	/opt/netbox/venv/bin/python3
Python Version:	3.7.3


In template /opt/netbox/netbox/templates/utilities/obj_table.html, error at line 27
'tuple' object has no attribute 'pk'

```

Further down in the debug page, I see this as last line:

```
/opt/netbox/netbox/utilities/tables.py, line 131, in <genexpr>
            self._objects_count = sum(1 for obj in self.data if getattr(obj, 'pk')) …
▼ Local vars
Variable	Value
.0	<list_iterator object at 0x7fe8ac30b9b0>
obj	(8, '192.168.202.1/24')
```
Interesting is, that the 192.168.202.1 IP does not exists. Not as /24 nor as /26

It seems that if the pagination kicks in, that the error is raised. If I set the page length to 25, it errors. When I set it to 500, it works fine.
I'm able to reproduce this on the demo.netbox.net

Who can help me out and has advise on where to start looking ?

Pieter

**Comment by PieterL75** [comment]
Actually,

It seems to be more generic.
Whenever you look at IP addresses of a prefix, and there is more than one page, the 'tuple' object has no attribute 'pk' error is raised
It works in v3.0.4
Maybe v3.0.5 had a change that introduced this error ?

Pieter

**Comment by kkthxbye-code** [comment, marked as answer]
Fixed here: https://github.com/netbox-community/netbox/commit/e09024e86f68dbd60f1c32b94abd6a64d54e7cc6

Will be in release 3.0.6.
